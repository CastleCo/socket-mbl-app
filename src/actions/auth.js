
import { AUTH_REGISTER_ASYNC, AUTH_LOGIN_ASYNC } from '../actionTypes/auth'

export function registerUser(email, password, firstName, lastName) {
  console.log("register action");
  return {
    type: AUTH_REGISTER_ASYNC,
    payload: {
      email,
      password,
      firstName,
      lastName
    }
  }
};

export function loginUser(email, password) {
  console.log("login action");
  return {
    type: AUTH_LOGIN_ASYNC,
    payload: {
      email,
      password,
    }
  }
};
