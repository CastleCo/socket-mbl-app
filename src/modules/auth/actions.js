import * as actionTypes from './action-types';

export const setUser = function (user) {
  return { type: actionTypes.SET_USER, user };
};

export const setTokens = function (refreshToken, accessToken) {
  return { type: actionTypes.SET_TOKENS, refreshToken, accessToken };
};

export const register = function(email, password, firstName, lastName) {
  return {
    type: actionTypes.REGISTER.REQUESTED,
    payload: { email, password, firstName, lastName }
  };
};

export const login = function (email, password) {
  return { type: actionTypes.LOGIN.REQUESTED, payload: { email, password } };
};

export const logout = function () {
  return { type: actionTypes.LOGOUT.REQUESTED };
};