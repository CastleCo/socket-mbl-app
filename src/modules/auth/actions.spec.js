import * as actions from "./actions";

describe("actions.setUser", () => {
  it("should match the snapshot", () => {
    expect(actions.setUser("USER")).toMatchSnapshot();
  });
});

describe("actions.setTokens", () => {
  it("should match the snapshot", () => {
    expect(actions.setTokens("ACCESS_TOKEN", "REFRESH_TOKEN")).toMatchSnapshot();
  });
});

describe("actions.login", () => {
  it("should match the snapshot", () => {
    expect(actions.login("USER_EMAIL", "USER_PASSWORD")).toMatchSnapshot();
  });
});

describe("actions.register", () => {
  it("should match the snapshot", () => {
    expect(actions.register("USER_EMAIL", "USER_PASSWORD", "USER_FIRSTNAME", "USER_LASTNAME")).toMatchSnapshot();
  });
});