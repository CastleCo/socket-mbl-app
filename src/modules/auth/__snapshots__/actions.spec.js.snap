// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions.login should match the snapshot 1`] = `
Object {
  "payload": Object {
    "email": "USER_EMAIL",
    "password": "USER_PASSWORD",
  },
  "type": "@@AUTH/LOGIN/REQUESTED",
}
`;

exports[`actions.register should match the snapshot 1`] = `
Object {
  "payload": Object {
    "email": "USER_EMAIL",
    "firstName": "USER_FIRSTNAME",
    "lastName": "USER_LASTNAME",
    "password": "USER_PASSWORD",
  },
  "type": "@@AUTH/REGISTER/REQUESTED",
}
`;

exports[`actions.setTokens should match the snapshot 1`] = `
Object {
  "accessToken": "REFRESH_TOKEN",
  "refreshToken": "ACCESS_TOKEN",
  "type": "@@AUTH/SET_TOKENS",
}
`;

exports[`actions.setUser should match the snapshot 1`] = `
Object {
  "type": "@@AUTH/SET_USER",
  "user": "USER",
}
`;
